### PR TITLE
ADD fixContentLengths.py

### PR DESCRIPTION
**Pull request by fgalan**
@iariasleon

### CODE REVIEW

**Comment by fgalan** [comment]
FYI: @crbrox @kzangeli this script can be very useful to avoid manual adjusting Content-Length in .test, e.g:

```
./fixContentLengths.py -v -f cases/0000_bad_input/erreneous_input.test
./fixContentLengths.py -v -f cases/0000_bad_input/
./fixContentLengths.py -v -f cases
```

File: test/functionalTest/fixContentLengths.py [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+    """
+    global verbose
+
+    if verbose:
+        print m
+
+
+def usage_and_exit(m):
+    """
+    Print usage message and exit"
```
Comment (kzangeli) [inline test/functionalTest/fixContentLengths.py]: Is this double-quote at EOL intended?
Looks funny ...

Comment (fgalan) [inline test/functionalTest/fixContentLengths.py]: No. It isn't. I'll fix it.

File: test/functionalTest/fixContentLengths.py [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+    Open file passed as argument, looking for Content-Lengh lines. A list of pairs is removed, the
+    first item in each pair being a line number and the second item the Content-Lenght value
+
+    :param file_name: file to be processed
+    :return: a list of pairs as described above
+    """
+
+    r = []
+    l = 0
+    for line in open(file_name):
```
Comment (iariasleon) [inline test/functionalTest/fixContentLengths.py]: It is recommend to use try-except to catch the exception

File: test/functionalTest/fixContentLengths.py [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+    # To be sure directory hasn't a trailing slash
+    dir_name_clean = dir_name.rstrip('/')
+
+    accum = [0, 0, 0]
+
+    for file in os.listdir(dir_name_clean):
+
+        file_name = dir_name_clean + '/' + file
+
+        if os.path.isdir(file_name):
```
Comment (iariasleon) [inline test/functionalTest/fixContentLengths.py]: It is recommend to use try-except to catch the exception

File: test/functionalTest/fixContentLengths.py [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+         if first item is True)
+       * accumulated difference in lengths between .out and .test (only makes sense if first item is True)
+    """
+
+    # To be sure directory hasn't a trailing slash
+    dir_name_clean = dir_name.rstrip('/')
+
+    accum = [0, 0, 0]
+
+    for file in os.listdir(dir_name_clean):
```
Comment (iariasleon) [inline test/functionalTest/fixContentLengths.py]: It is recommend in I/O to use try-except to catch the exception.

File: test/functionalTest/fixContentLengths.py (hunk 2 above)
Comment (fgalan) [inline test/functionalTest/fixContentLengths.py]: Fixed (for this an other `open()` ocurrence) in 5765b576b7b694e8b0c4b5f65a8dfb669f7a17c4

File: test/functionalTest/fixContentLengths.py (hunk 4 above)
Comment (fgalan) [inline test/functionalTest/fixContentLengths.py]: Fixed in 5765b576b7b694e8b0c4b5f65a8dfb669f7a17c4

**Comment by kzangeli** [comment]
LGTM

**Comment by iariasleon** [comment]
LGTM

**Pull request merged**